# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/PlatformEncoderModule.h
# Commit: ea4d0db67f89
# Full Hash: ea4d0db67f899714106e6f4ffcc78ebe8feaf466
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 1
# Description:
#   Bug 1749047 - Implement plumbing for SVC. r=chunmin,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D196473
# ==============================================================================

diff -r f0ac27321cf8 -r ea4d0db67f89 dom/media/platforms/PlatformEncoderModule.h
--- a/dom/media/platforms/PlatformEncoderModule.h	Fri Dec 22 21:39:18 2023 +0000
+++ b/dom/media/platforms/PlatformEncoderModule.h	Fri Dec 22 21:39:18 2023 +0000
@@ -150,6 +150,7 @@
   };
   using PixelFormat = dom::ImageBitmapFormat;
   enum BitrateMode { Constant, Variable };
+  enum ScalabilityMode { None, L1T2, L1T3 };
 
   enum class HardwarePreference { RequireHardware, RequireSoftware, None };
 
@@ -238,6 +239,7 @@
         mHardwarePreference(aConfig.mHardwarePreference),
         mPixelFormat(aConfig.mPixelFormat),
         mSourcePixelFormat(aConfig.mSourcePixelFormat),
+        mScalabilityMode(aConfig.mScalabilityMode),
         mFramerate(aConfig.mFramerate),
         mKeyframeInterval(aConfig.mKeyframeInterval),
         mBitrate(aConfig.mBitrate),
@@ -253,6 +255,7 @@
                 const uint32_t aBitrate,
                 const MediaDataEncoder::BitrateMode aBitrateMode,
                 const MediaDataEncoder::HardwarePreference aHardwarePreference,
+                const MediaDataEncoder::ScalabilityMode aScalabilityMode,
                 const Maybe<CodecSpecific>& aCodecSpecific)
       : mCodec(aCodecType),
         mSize(aSize),
@@ -260,6 +263,7 @@
         mHardwarePreference(aHardwarePreference),
         mPixelFormat(aPixelFormat),
         mSourcePixelFormat(aSourcePixelFormat),
+        mScalabilityMode(aScalabilityMode),
         mFramerate(aFramerate),
         mKeyframeInterval(aKeyframeInterval),
         mBitrate(aBitrate),
@@ -294,6 +298,7 @@
   MediaDataEncoder::HardwarePreference mHardwarePreference;
   MediaDataEncoder::PixelFormat mPixelFormat;
   MediaDataEncoder::PixelFormat mSourcePixelFormat;
+  MediaDataEncoder::ScalabilityMode mScalabilityMode;
   uint8_t mFramerate{};
   size_t mKeyframeInterval{};
   uint32_t mBitrate{};